src/DataGrid.tsx

```diff
@@ -487,21 +479,29 @@ function DataGrid<R, SR, K extends Key>(
   /**
    * callbacks
    */
-  const handleColumnResize = useCallback(
+  const handleColumnResize = useLatestFunc(
     (column: CalculatedColumn<R, SR>, width: number | 'max-content') => {
       if (width === 'max-content') {
-        setAutoResizeColumn(column);
+        flushSync(() => {
+          // TODO: only clear flex columns, if any?
+          setMeasuredColumnWidths(new Map());
```

Comment:
Should we only clear the width of flex columns that are in the viewport (`flexWidthViewportColumns`)? Would that fix the bug?

Reply:
I'm not sure about using viewport columns, as resizing may display columns that were previously hidden on the next render.